host/host.go

```diff
@@ -268,6 +268,17 @@ func (h *Host) PrivateListenMaddr() (ma.Multiaddr, error) {
 	return nil, fmt.Errorf("no private multi address found in %s", h.Addrs())
 }
 
+// LocalListenMaddr returns the first multiaddress in a localhost IP range that
+// this host is listening on.
+func (h *Host) LocalListenMaddr() (ma.Multiaddr, error) {
+	for _, maddr := range h.Addrs() {
+		if manet.IsIPLoopback(maddr) {
```

Comment:
Does it need to be specifically `ip4` or `ip6` or it doesn't matter and both addresses are acceptable?

Reply:
That is a good question. I've been checking at the [Prysm code](https://github.com/prysmaticlabs/prysm/blob/ae451a3a029d50fb6e46a76de5c9b9ea657465b1/beacon-chain/rpc/prysm/node/handlers.go#L79), and it doesn't seem to check anything, it just parses the `maddr` info using the `go-libp2p` primitives:
```go 
func (s *Server) AddTrustedPeer(w http.ResponseWriter, r *http.Request) {
     ...
     info, err := peer.AddrInfoFromString(addrRequest.Addr)
     ....
```
Do you think that could be a problem? We don't seem to be checking anything like that for the `private` maddr neither:
```go
// PrivateListenMaddr returns the first multiaddress in a private IP range that
// this host is listening on.
func (h *Host) PrivateListenMaddr() (ma.Multiaddr, error) {
	for _, maddr := range h.Addrs() {
		if manet.IsIPLoopback(maddr) {
			continue
		}
		if !manet.IsPrivateAddr(maddr) {
			continue
		}
		return maddr, nil
	}
	return nil, fmt.Errorf("no private multi address found in %s", h.Addrs())
}
```

Reply:
As long as both Hermes and Prysm have both an `ip4` and `ip6` and can reach each other on both stacks it is fine (or none of them is using `ip6`). Otherwise, we could also restrict both to use `ip4`.